src/_pytest/assertion/rewrite.py

```diff
@@ -466,10 +467,14 @@ def _saferepr(obj):
 
     """
     r = py.io.saferepr(obj)
-    if isinstance(r, six.text_type):
-        return r.replace(u"\n", u"\\n")
-    else:
-        return r.replace(b"\n", b"\\n")
+    # only occurs in python2.x, repr must return text in python3+
+    if isinstance(r, bytes):
+        # Represent unprintable bytes as `\x##`
+        r = u"".join(
+            u"\\x{:x}".format(ord(c)) if c not in string.printable else c.decode()
```

Comment:
isn't there the hex-escape encoding we can use instead?

Reply:
string-escape, yes: but it also escapes quote characters
unicode-escape: almost, but it decodes escape sequences to their unicode character

Reply:
@asottile i see, thanks for researching the details